test/integration/scheduler/framework_test.go

```diff
@@ -2289,15 +2168,41 @@ func initRegistryAndConfig(t *testing.T, pp ...*PermitPlugin) (frameworkruntime.
 	versionedCfg := v1beta3.KubeSchedulerConfiguration{
 		Profiles: []v1beta3.KubeSchedulerProfile{{
 			SchedulerName: pointer.StringPtr(v1.DefaultSchedulerName),
-			Plugins: &v1beta3.Plugins{
-				Permit: v1beta3.PluginSet{},
-			},
+			Plugins:       &v1beta3.Plugins{},
 		}},
 	}
 	registry = frameworkruntime.Registry{}
 	for _, p := range pp {
-		registry.Register(p.Name(), newPermitPlugin(p))
-		versionedCfg.Profiles[0].Plugins.Permit.Enabled = append(versionedCfg.Profiles[0].Plugins.Permit.Enabled, v1beta3.Plugin{Name: p.Name()})
+		// register plugin
+		switch pl := p.(type) {
+		case *PermitPlugin:
+			registry.Register(p.Name(), newPermitPlugin(pl))
+		default:
+			registry.Register(p.Name(), newPlugin(p))
+		}
+
```

Comment:
Make code cleaner?
```go
plugins := &v1beta3.Plugins{}
plugin := v1beta3.Plugin{Name: p.Name()}
switch p.(type) {
    .......
    plugins.PreFilter.Enabled = append(plugins.PreFilter.Enabled, plugin)
}

versionedCfg.Profiles[0].Plugins = plugins
```

Reply:
hmmm, I should review more carefully. Thanks Alex.